### ISSUE TITLE
Handle leading dimensions

### ISSUE THREAD

**Issue by awav** [GPflow/GPflow#1204]
If GPflow were able to handle leading dimensions that would allow running the training of multiple models in parallel. E.g. we could run several models with different initial parameters, maybe even with separate settings for transformation, etc. Also, It would give away for efficient MCMC runs in parallel chains!

**Comment by markvdw** [comment]
So I think this is cool, but rather low priority. We can already scale up multiple models just by adding more computation, e.g. running models in parallel on different machines.

Given that for large models the CPU+GPU are already utilised at near peak capacity, I expect that the speed-up will only be large when training many small models. In addition, I'm worried it may make the interface a lot more complicated.

**Comment by awav** [comment]
@markvdw, It is a part of the feature list in the project board. All tasks there are important to consider and discuss, an either low or high priority. The interface should not change at all. E.g. check TensorFlow probabilities, they handle any leading dimensions.
